tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyMain.java

```diff
@@ -3570,6 +3590,21 @@ public static Phase phase() {
         return currentPhase;
     }
 
+    public static Phase phase(CLDRLocale locale) {
+        return (SubmissionLocales.isTcLocale(locale) ? phase() : getDDLPhase());
+    }
+
+    /**
+     * @returns the current phase for the locale. This is the preferred API.
+     */
+    public static CheckCLDR.Phase getCPhase(CLDRLocale loc) {
+        return phase(loc).getCPhase();
```

Comment:
"C" for "current" is a little cryptic; I didn't know what it stood for until I got here. How about getCurrentPhase? Or a name that clarifies this method is appropriate for both DDL and non-DDL locales?

It seems like a terminology is needed, distinguishing 3 categories:

1.  methods/data specifically for DDL
2. methods/data specifically for non-DDL
3. methods/data that apply to either DDL or non-DDL, depending on the specified locale

There's also the long-standing confusion between CheckCLDR.Phase and SurveyMain.Phase, which still makes my head spin. Of course, fixing that is out of the scope of this ticket, but as long as new methods are being named, it would SO helpful for the new methods/data to use a naming convention that distinguishes them -- maybe checkPhase vs stPhase?

Also, it seems error prone to have three public methods: phase(), phase(locale), and getCPhase(loc). If the no-argument phase() ignores the existence of DDL, what ensures that various modules are consistent in treating DDL differently from non-DDL?

Reply:
it's not current, it's CheckCLDR.  CheckCLDR added its own 'Phase' enum distinct from ST's, and they haven't been fully reconciled yet. It's the "long-standing confusion between CheckCLDR.Phase and SurveyMain.Phase".

Reply:
I will work on making these more clearer and consistent.  Perhaps phase() could be renamed to overallPhase() or something.

Reply:
And that would need to be overallPhase(locale), right? Without the locale, it can't be overall. So ideally, phase() without a locale parameter might be private? currently it has about 14 callers (before this PR) (not counting jsp)